packages/smooth_app/lib/pages/product/common/product_query_page.dart

```diff
@@ -203,9 +202,12 @@ class _ProductQueryPageState extends State<ProductQueryPage>
           actions: _getAppBarButtons(),
         ),
         body: RefreshIndicator(
-          onRefresh: () => refreshList(),
+          onRefresh: () async {
+            await refreshList();
+          },
```

Comment:
What's wrong with arrows?
What about `onRefresh: () async => _refreshList()`?

Reply:
`=>` was directly returning the Future, and by putting it in brackets, Future can be awaited.

Reply:
Then try `async =>`

Reply:
Yes, I mean to say that `=>` this directly returns the `Future` instead of resolving it.

it's like typing 
```
()async{
   return Future;
}
```

Reply:
Jus to be sure: both the following codes do _exactly_ the same thing, don't they?

```dart
onRefresh: () async {
  await refreshList();
},
```

and

```dart
onRefresh: () async => refreshList(),
```

Reply:
This waits till the `refreshList` is completed, so onRefresh will be marked as complete only when this whole function executes fully.
> ```dart
> onRefresh: () async {
>   await refreshList();
> },
> ```

Here, onRefresh will show loaded state, even if `refreshList` is still executing as it doesn't await, it directly returns the value.
> 
> ```dart
> onRefresh: () async => refreshList(),
> ```

Both methods behave differently

Reply:
@anshpathania7 Interesting, but I'm not convinced. I'll double-check that tomorrow! I have to think of a test set-up.
What do you mean with "onRefresh _will show loaded state_"?

Reply:
> What do you mean with "onRefresh _will show loaded state_"?

In documentation it says that `onRefresh` 
must complete it's Future. So as long the Future is executing, it will show `refreshing indicator/ loading state`.

Suppose, if we directly return Future without using await, code will still execute but it wont show loading indicator to user.

Reply:
I rechecked it again, Apparantly I'm wrong😅, 

`() async =>` works as intended, but to use `try/catch`blocl ,  `()async{ }` will be needed.

Reply:
The try/catch should be inside the method, like in `_downloadNextPage()`.
So no problem with
`onRefresh: () async => refreshList()`